internal/db/schema/migrations/oss/postgres/29/04_kms_keys.up.sql

```diff
@@ -0,0 +1,215 @@
+begin;
+
+-- make the required schema changes to adopt
+-- github.com/hashicorp/go-kms-wrapping/extras/kms/v2 
+-- this migration is from 04_keys.up.sql in that repo. 
+
+
+-- we can use our existing kms_root_key table.
+
+-- we need to add an add additional constraint to the kms_root_key_version
+-- table: 
+-- https://github.com/hashicorp/go-kms-wrapping/blob/c06f9db9380f2c26cad07cf51f7a324d25dd55ba/extras/kms/migrations/postgres/04_keys.up.sql#L29 
+alter table kms_root_key_version
+  add constraint not_empty_key
+  check (
+      length(key) > 0
+    );
+
+
+-- we need to create kms_data_key and kms_data_key_version tables.  we will
+-- intentionally be using the boundary domain types and trigger functions. 
+create table kms_data_key (
+  private_id kms_private_id primary key,
+  root_key_id kms_private_id not null
+    references kms_root_key(private_id)
+    on delete cascade
+    on update cascade,
+  purpose text not null
+    constraint not_start_end_whitespace_purpose
+    check (length(trim(purpose)) = length(purpose)),
+  create_time kms_timestamp,
+  unique (root_key_id, purpose) -- there can only be one dek for a specific purpose per root key
+);
+comment on table kms_data_key is
+  'kms_data_key contains deks (data keys) for specific purposes';
+
+ -- define the immutable fields for kms_data_key (all of them)
+create trigger immutable_columns
+before
+update on kms_data_key
+  for each row execute procedure kms_immutable_columns('private_id', 'root_key_id', 'purpose', 'create_time');
+
+-- not adding the kms_default_create_time_column trigger until after we've
+-- converted the data (we don't want the timestamp to change)
+
+create table kms_data_key_version (
+  private_id kms_private_id primary key,
+  data_key_id kms_private_id not null
+    references kms_data_key(private_id) 
+    on delete cascade 
+    on update cascade, 
+  root_key_version_id kms_private_id not null
+    references kms_root_key_version(private_id) 
+    on delete cascade 
+    on update cascade,
+  version kms_version,
+  key bytea not null
+    constraint not_empty_key
+    check (
+      length(key) > 0
+    ),
+  create_time kms_timestamp,
+  unique(data_key_id, version)
+);
+comment on table kms_data_key is
+  'kms_data_key_version contains versions of a kms_data_key (dek aka data keys)';
+
+ -- define the immutable fields for kms_data_key_version (all of them)
+create trigger immutable_columns
+before
+update on kms_data_key_version
+  for each row execute procedure immutable_columns('private_id', 'data_key_id', 'root_key_version_id', 'version', 'key', 'create_time');
```

Comment:
Similarly I think this should be `kms_immutable_columns` and the procedure that is executed should be `kms_immutable_columns`

Reply:
ty